Error: Command failed: npm run build

for command  `npm init "@observablehq"` on windows 11 with the latest node.js getting below error:
```
|
o  Initialize git repository?
|  Yes
|
o  Initializing Framework cache...|
x  Error: Command failed: npm run build
|  (node:24144) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
|  (Use `node --trace-deprecation ...` to show where the warning was created)
|
|  Unexpected error: fetch failed
|
|  Tip: To see the full stack trace, run with the --debug flag.
|
|  If you think this is a bug, please file an issue at
|  ↳ https://github.com/observablehq/framework/issues
|
|
|
•  To see the full stack trace, run with the --debug flag.
│
│  If you think this is a bug, please file an issue at
└  https://github.com/observablehq/framework/issues

x  Canceled
npm ERR! code 1
npm ERR! path C:\Users\sonalib
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c observablehq-create

npm ERR! A complete log of this run can be found in: C:\Users\sonalib\AppData\Local\npm-cache\_logs\2024-04-02T10_18_23_049Z-debug-0.log
```

--- a/src/bin/observable.ts
+++ b/src/bin/observable.ts
@@ -242,8 +242,8 @@ try {
         // clack.outro doesn't handle multiple lines well, so do it manually
         console.log(
           `${faint("│\n│")}  If you think this is a bug, please file an issue at\n${faint("└")}  ${link(
-            "https://github.com/observablehq/framework/issues\n"
-          )}`
+            "https://github.com/observablehq/framework/issues"
+          )}\n`
         );
       }
     } else {
@@ -255,8 +255,8 @@ try {
         console.error("\nTip: To see the full stack trace, run with the --debug flag.\n");
         console.error(
           `If you think this is a bug, please file an issue at\n↳ ${link(
-            "https://github.com/observablehq/framework/issues\n"
-          )}`
+            "https://github.com/observablehq/framework/issues"
+          )}\n`
         );
       }
     }

--- a/src/create.ts
+++ b/src/create.ts
@@ -8,24 +8,23 @@ import {fileURLToPath} from "node:url";
 import {promisify} from "node:util";
 import * as clack from "@clack/prompts";
 import untildify from "untildify";
+import wrapAnsi from "wrap-ansi";
 import type {ClackEffects} from "./clack.js";
-import {cyan, faint, inverse, link, reset} from "./tty.js";
+import type {TtyEffects} from "./tty.js";
+import {cyan, defaultEffects as defaultTtyEffects, faint, inverse, link, reset} from "./tty.js";
 
-export interface CreateEffects {
+export interface CreateEffects extends TtyEffects {
   clack: ClackEffects;
   sleep: (delay?: number) => Promise<void>;
-  log(output: string): void;
   mkdir(outputPath: string, options?: {recursive?: boolean}): Promise<void>;
   copyFile(sourcePath: string, outputPath: string): Promise<void>;
   writeFile(outputPath: string, contents: string): Promise<void>;
 }
 
 const defaultEffects: CreateEffects = {
+  ...defaultTtyEffects,
   clack,
   sleep,
-  log(output: string): void {
-    console.log(output);
-  },
   async mkdir(outputPath: string, options): Promise<void> {
     await mkdir(outputPath, options);
   },
@@ -82,6 +81,7 @@ export async function create(effects: CreateEffects = defaultEffects): Promise<v
         }),
       installing: async ({results: {rootPath, projectTitle, includeSampleFiles, packageManager, initializeGit}}) => {
         rootPath = untildify(rootPath!);
+        let spinning = true;
         const s = clack.spinner();
         s.start("Copying template files");
         const template = includeSampleFiles ? "default" : "empty";
@@ -114,9 +114,20 @@ export async function create(effects: CreateEffects = defaultEffects): Promise<v
         }
         if (packageManager) {
           s.message("Initializing Framework cache");
-          await promisify(exec)(`${runCommand} build`, {cwd: rootPath});
+          try {
+            await promisify(exec)(`${runCommand} build`, {cwd: rootPath});
+          } catch {
+            spinning = false;
+            s.stop("Installed! 🎉");
+            clack.log.warn(
+              wrapAnsi(
+                "Failed to initialize Framework cache. This may be a transient error loading data from external servers or downloading imported modules from jsDelivr; or it might be a network configuration issue such as a firewall blocking traffic. You can ignore this error for now and Framework will automatically try to download again on preview or build. If you continue to experience issues, please check your network configuration.",
+                Math.min(80, effects.outputColumns)
+              ) + `\n\nWant help? ${link("https://github.com/observablehq/framework/issues")}\n`
+            );
+          }
         }
-        s.stop("Installed! 🎉");
+        if (spinning) s.stop("Installed! 🎉");
         const instructions = [`cd ${rootPath}`, ...(packageManager ? [] : [installCommand]), `${runCommand} dev`];
         clack.note(instructions.map((line) => reset(cyan(line))).join("\n"), "Next steps…");
         clack.outro(`Problems? ${link("https://observablehq.com/framework/getting-started")}`);

--- a/test/create-test.ts
+++ b/test/create-test.ts
@@ -4,6 +4,7 @@ import type {CreateEffects} from "../src/create.js";
 import {create} from "../src/create.js";
 import {fromOsPath} from "../src/files.js";
 import {TestClackEffects} from "./mocks/clack.js";
+import {MockLogger} from "./mocks/logger.js";
 
 describe("create", () => {
   it("instantiates the default template", async () => {
@@ -60,10 +61,12 @@ describe("create", () => {
 });
 
 class TestCreateEffects implements CreateEffects {
+  isTty = true;
+  outputColumns = 80;
+  logger = new MockLogger();
   outputs = new Map<string, string>();
   clack = new TestClackEffects();
   async sleep(): Promise<void> {}
-  log(): void {}
   async mkdir(): Promise<void> {} // TODO test?
   async copyFile(sourcePath: string, outputPath: string): Promise<void> {
     this.outputs.set(fromOsPath(outputPath), await readFile(sourcePath, "utf-8"));

